.github/workflows/ci-test-ruby.yaml

```diff
@@ -0,0 +1,76 @@
+name: Test Ruby
+
+on:
+  workflow_call:
+    inputs:
+      extraSystemDependencies:
+        description: 'Install additional system dependencies'
+        required: false
+        default: ''
+        type: string
+
+jobs:
+  test-ruby:
+    name: Test Ruby
+    runs-on: ubuntu-latest
+    services:
+      mysql:
+        image: mysql:8.0
+        env:
+          MYSQL_ROOT_PASSWORD: root
+        ports:
+          - 3306:3306
+        options: >-
+          --health-cmd "mysqladmin ping"
+          --health-start-period 10s
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      redis:
+        image: redis
```

Comment:
optional: consider redis:alpine (should be slightly quicker and use slightly less resources)